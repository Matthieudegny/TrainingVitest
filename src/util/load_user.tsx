interface User {
  username: string;
  name: string;
  projects: string[];
  coolness?: number;
}

async function loadUser(username: string): Promise<User | undefined> {
  const users: User[] = [
    {
      username: "antho",
      name: "Bob",
      projects: ["vitest", "vite"],
    },
    {
      username: "matt",
      name: "Zary",
      projects: ["react", "solidity"],
    },
  ];

  return users.find((user) => user.username === username);
}

export async function loadUserData(username: string) {
  const user = await loadUser(username);
  if (!user) throw new Error("no user found");
  user.coolness = username === "antho" ? 100 : -1;
  return user;
}
