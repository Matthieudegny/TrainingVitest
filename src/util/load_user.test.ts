import { describe, test, expect } from "vitest";
import { loadUserData } from "./load_user.tsx";

describe("loadUserDetails", () => {
  //test v1
  //   test("load user data as expected", async () => {
  //     const user = await loadUserData("antho");

  //     expect(user).toEqual({
  //       coolness: 100,
  //       name: "Bob",
  //       projects: ["vitest", "vite"],
  //       username: "antho",
  //     });
  //   });

  //testV2
  test("load user data as expected", async () => {
    const user = await loadUserData("antho");
    expect(user).toMatchSnapshot();
  });
  test("set coolness level appropriately", async () => {
    const antho = await loadUserData("antho");
    const matt = await loadUserData("matt");

    expect(antho.coolness).toBe(100);
    expect(matt.coolness).toBe(-1);
  });
  test("throws an error for noneexistant users", async () => {
    //directement la fonciton dans le expect car le throw new error dans le code de la fonction
    //est donnÃ© avant la fin de la fonction
    expect(async () => await loadUserData("fakeuser")).rejects.toThrowError("no user found");
  });
});
