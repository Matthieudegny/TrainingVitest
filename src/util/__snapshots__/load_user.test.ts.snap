// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loadUserDetails > load user data as expected 1`] = `
{
  "coolness": 100,
  "name": "Bob",
  "projects": [
    "vitest",
    "vite",
  ],
  "username": "antho",
}
`;
